lib/jellyfish/notifier.ex

```diff
@@ -148,30 +204,54 @@ defmodule Jellyfish.Notifier do
     state = %{
       caller_pid: self(),
       subscriptions: %{all: MapSet.new()},
-      pending_subscriptions: %{}
+      pending_subscriptions: %{},
+      subscribed_events: []
     }
 
-    auth_msg =
-      %ServerMessage{content: {:auth_request, %AuthRequest{token: api_token}}}
+    with {:ok, events} <- Keyword.get(opts, :events, [:server_notification]) |> validate_events(),
+         state = %{state | subscribed_events: events},
+         {:ok, ws} <-
+           apply(WebSockex, fun, ["#{address}/socket/server/websocket", __MODULE__, state]),
+         {:jellyfish, :authenticated} <-
+           send_request(ws, {:auth_request, %AuthRequest{token: api_token}}),
+         {:jellyfish, :subscribed} <-
+           send_request(ws, {:subscribe_request, %SubscribeRequest{event_types: events}}) do
+      {:ok, ws}
+    else
+      {:jellyfish, :invalid_token} ->
+        {:error, :invalid_token}
+
+      {:error, _reason} = error ->
+        error
+    end
+  end
+
+  defp validate_events(events) do
+    with true <- Enum.all?(events, fn event -> event in @valid_events end) do
+      events
+      |> Enum.uniq()
+      |> Enum.map(&to_proto_event_type/1)
+      |> then(&{:ok, &1})
+    else
+      false ->
+        {:error, :invalid_event_type}
+    end
```

Comment:
```suggestion
    if Enum.all?(events, fn event -> event in @valid_events end) do
      events
      |> Enum.uniq()
      |> Enum.map(&to_proto_event_type/1)
      |> then(&{:ok, &1})
    else
      {:error, :invalid_event_type}
    end
```
Nitpick but we're not doing anything else in the `with` statement